packages/loader/container-definitions/src/deltas.ts

```diff
@@ -198,3 +201,16 @@ export interface IDeltaQueue<T> extends IEventProvider<IDeltaQueueEvents<T>>, ID
      */
     toArray(): T[];
 }
+
+export enum ReadOnlyType {
+    /** read-only type unknown because connection has not yet been established */
+    Unknown = "unknown",
+    /** write */
+    NotReadOnly = "notReadOnly",
+    /** read-only because client does not have write permissions for document */
+    ReadOnlyPermissions = "readOnlyPermissions",
+    /** read-only because forceReadOnly() was called */
+    ReadOnlyForced = "readOnlyForced",
+    /** read-only with no delta stream connection */
+    StorageOnly = "storageOnly",
```

Comment:
Something to consider: The way it's structured right now with enum means that caller can only learn one bit of information. I.e. we can only communicate ReadOnlyPermissions or ReadOnlyForced but not both. And client needs to check for Unknown & NotReadOnly if it can write to a file.
Maybe better structure would be something like that:
```
export type ReadOnlyInfo = {
    readonly readonly: false | undefined;
} | {
    readonly readonly: true;
    readonly forced: boolean;
    readonly permissions: boolean;
    readonly storageOnly: boolean;
};
```
I.e. provide all the data we know about, and let the client have full info, but also much easier to make trivial decisions (like can I change this file).